unleash-yggdrasil/src/lib.rs

```diff
@@ -528,26 +528,22 @@ impl EngineState {
         name: &str,
         context: &Context,
         external_values: &Option<HashMap<String, bool>>,
-    ) -> VariantDef {
+    ) -> ExtendedVariantDef {
         let toggle = self.get_toggle(name);
+        let enabled = toggle
+            .map(|t| self.enabled(t, context, external_values))
+            .unwrap_or_default();
 
-        let variant = if let Some(toggle) = toggle {
-            let enabled = self.enabled(toggle, context, external_values);
-            self.count_toggle(name, enabled);
-
-            if enabled {
-                self.check_variant_by_toggle(toggle, context)
-            } else {
-                None
-            }
-        } else {
-            self.count_toggle(name, false);
-            None
+        let variant = match toggle {
```

Comment:
Does the same as the block it removes,(minus the count which is now handled after this block).

This was getting a bit ugly and since we're in this space...